app/assets/javascripts/media_player_wrapper/mejs4_plugin_hd_toggle.es6

```diff
@@ -0,0 +1,190 @@
+
+// Feature configuration
+Object.assign(mejs.MepDefaults, {
+  /**
+   * @type {String}
+   */
+  hdToggleLabel: 'HD',
+  /**
+   * @type {String}
+   */
+  hdToggleText: null,
+  /**
+   * @type {String}
+   */
+  hdToggleOn: true,
+  /**
+   * @type {String}
+   */
+  hdToggleBetween: ["high", "medium"]
+});
+
+Object.assign(MediaElementPlayer.prototype, {
+  // Public variables (also documented according to JSDoc specifications)
+
+  /**
+   * Feature constructor.
+   *
+   * Always has to be prefixed with `build` and the name that will be used in MepDefaults.features list
+   * @param {MediaElementPlayer} player
+   * @param {HTMLElement} controls
+   * @param {HTMLElement} layers
+   * @param {HTMLElement} media
+   */
+  buildhdToggle(player, controls, layers, media) {
+    const t = this;
+    const hdToggleText = mejs.Utils.isString(t.options.hdToggleText) ? t.options.hdToggleText : mejs.i18n.t('mejs.hd-toggle')
+    player.qualities = [];
+
+    player.sources = $(player.domNode).find("source");
+    for (var i = 0; i < player.sources.length; i++) {
+      var src = player.sources[i];
+      for (var j = 0; j < player.options.hdToggleBetween.length; j++) {
+        if (src.getAttribute("data-quality") == player.options.hdToggleBetween[j]) {
+          player.qualities[j] = src.getAttribute("src");
+        }
+      }
+    }
+
+    player.hdtoggleButton = document.createElement('div');
+    player.hdtoggleButton.className =
+      t.options.classPrefix +
+      'button ' +
+      t.options.classPrefix +
+      'hd-toggle-button';
+    player.hdtoggleButton.innerHTML = `<button type="button" aria-controls="${ t.id }"
+      title="${ hdToggleText }">${ player.options.hdToggleLabel }</button>`;
+
+    player.hdtoggleButton.addEventListener(
+      'click',
+      t.toggleQuality.bind(t)
+    );
+
+    // Add control button to player
+    t.addControlElement(player.hdtoggleButton, 'hdToggle');
+
+    if (player.options.hdToggleOn && player.qualities[0] != null) {
+      player.hdtoggleButton.className += " mejs__hdtoggle-on";
+      player.switchStream(player.qualities[0]);
+    } else if (player.qualities[1] != null) {
+      player.switchStream(player.qualities[1]);
+    } else {
+      alert("Did not find " + hdToggleBetween[0] + " and " + hdToggleBetween[1] + " streams");
+    }
+  },
+
+  /**
+   * Feature destructor.
+   *
+   * Always has to be prefixed with `clean` and the name that was used in MepDefaults.features list
+   * @param {MediaElementPlayer} player
+   * @param {HTMLElement} controls
+   * @param {HTMLElement} layers
+   * @param {HTMLElement} media
+   */
+  cleanhdToggle(player, controls, layers, media) {
+    if (player && player.hdtoggleButton) {
+        player.hdtoggleButton.remove();
+      }
+  },
+
+  // Other optional public methods (all documented according to JSDoc specifications)
+
+  toggleQuality: function() {
+    var btn = $(this.hdtoggleButton);
```

Comment:
I mentioned this to Brian previously... don't change this now, but in the future we should prefix jQuery variables with a "$" for clarity.  For example `var btn = $(this.hdtoggleButton)` becomes `let $btn = $(this.hdtoggleButton`).